flag_test.go

```diff
@@ -1310,12 +1312,26 @@ func TestJSONGenericWrapper(t *testing.T) {
 				return &m
 			}(),
 		},
+		{
+			name: "json-incompatible-map",
+			wrapped: func() *map[string]interface{} {
+				m := map[string]interface{}{
+					"neighbors": []float64{11, math.Inf(1)},
+				}
+				return &m
+			}(),
+			expectErr: true,
+		},
 	} {
 		t.Run(tc.name, func(ct *testing.T) {
 			fl := JSONWrapGeneric(tc.wrapped)
 
 			if err := fl.Set(fl.String()); err != nil {
```

Comment:
if err == nil and expectErr is true that is a t.Fatal(...). so these kinds of checks have 2 parts

if err != nil && !expectErr {

} else if err == nil && expectErr {

}